gdx/src/com/badlogic/gdx/utils/Pools.java

```diff
@@ -16,13 +16,16 @@
 
 package com.badlogic.gdx.utils;
 
-/** Stores a map of {@link Pool}s (usually {@link ReflectionPool}s) by type for convenient static access.
+import com.badlogic.gdx.utils.DefaultPool.PoolSupplier;
+
+/** Stores a map of {@link Pool}s by type for convenient static access.
  * @author Nathan Sweet */
 public class Pools {
 	static private final ObjectMap<Class, Pool> typePools = new ObjectMap();
 
 	/** Returns a new or existing pool for the specified type, stored in a Class to {@link Pool} map. Note the max size is ignored
 	 * if this is not the first time this pool has been requested. */
+	@Deprecated
```

Comment:
> For one thing, get(Class) is just an easier API to call.

I agree and I struggled a bit with the `Pools` class. On the one hand I wanted to keep the semantics of the old methods with a clear replacement, on the other hand it's not easily possible.
If I could re-desing the class I would split the logic up to a `Pools#create` and `Pools#get/obtain` methods. And before you can call `get/obtain` you need to create/set a Pool first one time. This would be in my opinion an okay migration requirement and should be easy enough when well documented.

> Would it be possible to have ReflectionPool actually be a DefaultPool that acquires a constructor via reflection?

Yes, but I don't really like the idea. Preferably we could offer a good enough `DefaultPool` API that makes the `ReflectionPool` uneccessary. 

> I don't think this should be merged without explaining in the docs what the deprecation is related to, and how to get past it.

Agreed, forgot to document it.

> Also, obtain() now calls what is deprecated API.

Yes, oversaw that.

Reply:
I'd go as far to say any gains from the "easier" usage api will be lost by increased difficulty adding GWT/R8/Proguard/NativeImage support later.

I would go as far to say ReflectionPool should be entirely removed.